config/config.go

```diff
@@ -1019,6 +1019,9 @@ func (c *RelabelConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	if err := checkOverflow(c.XXX, "relabel_config"); err != nil {
 		return err
 	}
+	if (c.Regex == Regexp{}) {
```

Comment:
Another approach: Leave out the `omitempty` in L997. Does that do the trick?

It's kind of weird to first tell the marshaler to ignore an empty regex field to then check for it and create it after all.